benchmark_runner/jupyterlab/templates/summary_report/summary_report_operations.py

```diff
@@ -0,0 +1,353 @@
+            self.geometric_mean_df = pd.concat([geometric_mean_df_iops, geometric_mean_df_latency], ignore_index=True)
+        elif workload == 'bootstorm' or workload == 'windows':
+            self.geometric_mean_df = pd.DataFrame(self.geometric_mean_data)
+            self.median_indices = self.geometric_mean_df.groupby(['metric', 'ocp_version']).apply(self.__get_median)
+            self.geometric_mean_df = self.geometric_mean_df.loc[self.median_indices['median_result']]
+            self.geometric_mean_df = self.__calc_percentage(self.geometric_mean_df, complementary=True)
+            # updates metric names
+            if workload == 'windows':
+                windows_os = ['windows10', 'windows11', 'windows_server_2019', 'windows_server_2022']
+                for os in windows_os:
+                    self.geometric_mean_df['metric'] = self.geometric_mean_df['metric'].replace({os: f'p90 ({os})'})
+            else:
+                self.geometric_mean_df['metric'] = self.geometric_mean_df['metric'].replace(
+                    {'fedora37': 'p90 (fedora37)'})
+
+    @typechecked()
+    def get_median_geometric_mean_df(self, workload: str, df: pd.DataFrame):
+        """
+        This method calculates the geometric mean per workload
+        @param workload:
+        @param df:
+        @return:
+        """
+
+        logger.info('Wait till fetch and analyzing data ...')
+        # Convert lists to strings in the 'uuid' column
+        df['uuid'] = df['uuid'].apply(
+            lambda x: x[0] if isinstance(x, list) and x else x)
+
+        df = self.prepare_workload_dataframe(workload, df)
+
+        # Factorize the 'uuid' column
+        labels, uniques = pd.factorize(df['uuid'])
+
+        # Check if labels cover all rows
+        if len(labels) != len(df):
+            raise ValueError("Factorization did not cover all rows.")
+
+        # Calculate geometric mean for each unique 'uuid'
+        for label, unique_uuid in enumerate(uniques):
+            # logger.info(f"For each UUID {unique_uuid}:")
+            subset = df[labels == label]
+            if workload == 'hammerdb':
+                # Skip empty columns and skip NaN tpm values
+                if not subset.empty and subset['tpm'].notna().all():
+                    self.update_hammerdb_data(subset, unique_uuid)
+            elif workload == 'uperf':
+                if not subset.empty and subset['norm_ltcy'].notna().all() and subset['norm_byte'].notna().all():
+                    self.update_uperf_data(subset, unique_uuid)
+            elif workload == 'vdbench':
+                if not subset.empty and subset['Rate'].notna().all() and subset['Resp'].notna().all():
+                    self.update_vdbench_data(subset, unique_uuid)
+            elif workload == 'bootstorm' or workload == 'windows':
+                if not subset.empty and subset['bootstorm_time'].notna().all():
+                    self.update_bootstorm_data(subset, unique_uuid)
```

Comment:
Again, these really should be delegated.

Reply:
I dont see any reason to delegate it, its a different line code for each workload

Reply:
I mean that each workload should have its own function to do this.

Reply:
That what is exist, I have updated separate method for each workload